src/hotspot/share/oops/constantPool.hpp

```diff
@@ -108,10 +113,99 @@ class BSMAttributeEntry {
   int argument_count() const {
     return _argument_count;
   }
-  int argument_index(int n) const {
+  int argument(int n) const {
     assert(checked_cast<u2>(n) < _argument_count, "oob");
     return argument_indexes()[n];
   }
+
+  void set_argument(int index, u2 value) {
+    assert(index >= 0 && index < argument_count(), "invariant");
+    argument_indexes()[index] = value;
+  }
+};
+
+// The BSMAttributeEntries stores the state of the BootstrapMethods attribute.
+class BSMAttributeEntries {
+  friend class VMStructs;
+  friend class JVMCIVMStructs;
+
+public:
+  class InsertionIterator {
+    friend BSMAttributeEntries;
+    BSMAttributeEntries* insert_into;
+    // Current unused offset into BSMAEs offset array
+    int _cur_offset;
+    // Current unused offset into BSMAEs bsm-data array
+    int _cur_array;
+  public:
+    InsertionIterator() : insert_into(nullptr), _cur_offset(-1), _cur_array(-1) {}
+    InsertionIterator(BSMAttributeEntries* insert_into, int cur_offset, int cur_array)
+    : insert_into(insert_into),
+      _cur_offset(cur_offset),
+      _cur_array(cur_array) {}
```

Comment:
Not sure what the indentation rules are for constructors but I think we need some more.

Reply:
> Not sure what the indentation rules are for constructors

Same. I added 2 spaces.